### ISSUE TITLE
Consider exposing `WhoAmI` as a property of `Device`

### ISSUE THREAD

**Issue by bruno-f-cruz** [bonsai-rx/harp#83]
The `Device` constructor overload that takes a `WhoAmI` is not exposed in the Bonsai operator.
https://github.com/bonsai-rx/harp/blob/affa52f40fad128c033857b1aac3b27d7b7d9200/Bonsai.Harp/Device.cs#L38

The requirement to expose this WhoAmI was somewhat made mute by the device-specific APIs that implement the WhoAmI check on connection since these know what ID to expect to find.

There is however a scenario where having the WhoAmI exposed in the core class still makes some sense. Consider a user who has some sort of specification that maps `ComPort -> WhoAmI`.
We can then use `ComPort` to instantiate Observables that connect to different devices on different ports. This is already possible and an efficient way to connect to harp devices dynamically. The caveat here is that the WhoAmI information cannot currently be used to make check if the device we connected to is the expected device.  If this property is exposed, we can leverage the already existing constructor to have this behavior for free.

The proposal is thus to expose the WhoAmI property in the current operator as a `Nullable<int>` that by default will be null and ensure backward compatibility. If a value is defined, the aforementioned constructor will be used and the `WhoAmI` code will be asserted.

**Comment by glopesdev** [comment]
A workflow can already get the check, just not as a precondition. Connecting with `Device` produces the register dump, so filtering `WhoAmI` out of it and asserting gives the same guarantee one step later than the constructor would. That covers the `ComPort -> WhoAmI` mapping case without new surface.

Exposing the property is also less free than it looks. `deviceId` is readonly and set by the constructor, and generated devices pass their own identifier through `base(whoAmI)`. A settable `WhoAmI` on the base class would be settable on every generated subclass, letting a caller override the identity assertion a generated device makes about itself. Preventing that needs shadowing or per-subclass suppression, which is more design than the proposal assumes.

A new public property on `Device` is also device surface, and that is on hold while the `Harp.*` package split is being worked out, since whatever shape identity takes there should settle first.

Worth reopening if the runtime-identifier case comes up concretely rather than as a possibility, or once the package split lands and there is a natural home for it.
